code/game/objects/items/storage/storage.dm

```diff
@@ -445,7 +449,17 @@
 		I.mouse_opacity = initial(I.mouse_opacity)
 
 	update_icon()
-	return TRUE
+	return removal_good
+
+/obj/item/storage/proc/remove_gun_from_storage(obj/item/weapon/gun/gun)
```

Comment:
don't use usr, you have to get the mob/user from parent proc (and from attack_hand it seems)

and just inline this in remove_from_storage , it doesn't need to be that far up in the proc

Reply:
Another note, new procs always must be autodoced. See pins in dev channel